apps/api/src/app/widgets/usecases/mark-all-message-as-read-by-feed/mark-all-message-as-read-by-feed.usecase.ts

```diff
@@ -0,0 +1,62 @@
+import { Inject, Injectable } from '@nestjs/common';
+import { MessageRepository, SubscriberRepository } from '@novu/dal';
+import { AnalyticsService } from '@novu/application-generic';
+
+import { CacheKeyPrefixEnum } from '../../../shared/services/cache';
+import { QueueService } from '../../../shared/services/queue';
+import { ANALYTICS_SERVICE } from '../../../shared/shared.module';
+
+import { InvalidateCache } from '../../../shared/interceptors';
+import { MarkAllMessageAsReadByFeedCommand } from './mark-all-message-as-read-by-feed.command';
+import { ApiException } from '../../../shared/exceptions/api.exception';
+
+@Injectable()
+export class MarkAllMessageAsReadByFeed {
+  constructor(
+    private messageRepository: MessageRepository,
+    private queueService: QueueService,
+    private subscriberRepository: SubscriberRepository,
+    @Inject(ANALYTICS_SERVICE) private analyticsService: AnalyticsService
+  ) {}
+
+  @InvalidateCache([CacheKeyPrefixEnum.MESSAGE_COUNT, CacheKeyPrefixEnum.FEED])
+  async execute(command: MarkAllMessageAsReadByFeedCommand): Promise<number> {
+    const feedId = command.feedId ? [command.feedId] : null;
+    const subscriber = await this.subscriberRepository.findBySubscriberId(command.environmentId, command.subscriberId);
+    if (!subscriber) {
+      throw new ApiException(
+        `Subscriber ${command.subscriberId} does not exist in environment ${command.environmentId}, ` +
+          `please provide a valid subscriber identifier`
+      );
+    }
+    const response = await this.messageRepository.markAllUnreadAsReadByFeed(
+      subscriber._id,
+      command.environmentId,
+      feedId
+    );
+
+    this.queueService.wsSocketQueue.add({
+      event: 'unseen_count_changed',
+      userId: command.subscriberId,
```

Comment:
Notice you are using subscriber._id to get count and command.subscriberId to update socket. I do not think they are the same. 
I think you need to use subscriber._id. 
Maybe that's the reason your unseen_count was not updated?

Reply:
@ainouzgali, Thanks, it fixed it :D